changes as per Adrian's comments. (#180)
casper-network/docs-new: source/docs/casper/developers/writing-onchain-code/simple-contract.md

Reviewer: Let's add a call function snippet and the instruction on what goes in. Then at the end let's show the final result.

The project structure will be different while designing the full stack architecture. This will be expanded upon while describing the dApps.

### Creating the Project Automatically

> The `cargo casper` command can automatically set up the project structure, as shown [here](./getting-started.md#creating-a-project). Alternatively, follow the steps below to customize the project, yet create the various folders using `cargo`.

Reviewer: The link takes me to the other tutorial where I can't even see the structure. We should remove the link or point people to the proper place.

Reviewer: Let's use '#' to scroll up to see the structure. And change the word 'here' to 'above'

2. Create a folder for the contract code inside the project directory. This folder contains the logic that will be compiled into Wasm and executed on a Casper node. In this example, we named the folder `contract`. You can use a different folder name if you wish.

   - In the `contract` folder, add a source folder called `src` and a `Cargo.toml` file, which specifies the contract's dependencies.
   - Add a Rust file with the contract code in the `src` folder. In this example, we have the `main.rs` file.

3. Navigating back to the project directory, create a folder for the tests, which help verify the contract's functionality. In this example, we named the folder `tests`.

   - In the `tests` folder, add a source folder called `src` and a `Cargo.toml` file, which specifies the required dependencies to run the tests.
   - In the `src` folder, add a Rust file with the tests that verify the contract's behavior. In this example, we have the `integration-tests.rs` file.
4. Manually create Makefile and rust-toolchain as per [Semi-automatic setup (4.)](#semi-automatic-project-setup)
### Dependencies

> The `Cargo.toml` file includes the dependencies and versions the contract requires. At a minimum, you need to import the latest versions of the [casper-contract](https://docs.rs/casper-contract/latest/casper_contract/) and [casper-types](https://docs.rs/casper-types/latest/casper_types/) crates. The following dependencies and version numbers are only examples and must be adjusted based on your requirements.

Reviewer: Just to be more specific
```suggestion
The `Cargo.toml` file in the `contract` folder includes the dependencies and versions the contract requires. At a minimum, you need to import the latest versions of the [casper-contract](https://docs.rs/casper-contract/latest/casper_contract/) and [casper-types](https://docs.rs/casper-types/latest/casper_types/) crates. The following dependencies and version numbers are only examples and must be adjusted based on your requirements.
```

If you followed the [automatic setup](#automatic-project-setup), the dependencies should already be defined in `Cargo.toml`. For the [semi-automatic setup](#semi-automatic-project-setup) and [manual setup](#manual-project-setup) however, you'll need to manually add the dependencies to your crate's `Cargo.toml` file:

```toml
[dependencies]
# A library for developing Casper network smart contracts.
casper-contract = "1.4.4"
# Types shared by many Casper crates for use on a Casper network.
casper-types = "1.5.0"
```

- `casper-contract = "1.4.4"` - Provides the SDK for the execution engine (EE). The latest version of the crate is published [here](https://crates.io/crates/casper-contract).
- `casper-types = "1.5.0"` - Includes types shared by many Casper crates for use on a Casper network. This crate is necessary for the EE to understand and interpret the session code. The latest version of the crate is published [here](https://crates.io/crates/casper-types).

## Writing a Basic Smart Contract

> At this point you either have the default example contract defined in `contract/main.rs` ([automatic](#automatic-project-setup) setup using cargo-casper), an empty `contract/main.rs` file ([manual](#manual-project-setup) project setup), or a rust "hello world" program defined in your `contract/main.rs` ([semi-automatic](#semi-automatic-project-setup) setup using "vanilla cargo"). In the following, you will write a new contract step-by-step. Therefore it is recommended to clear the content of contract/main.rs ( if any ).

Reviewer: Removing unwanted spaces
```suggestion
At this point you either have the default example contract defined in `contract/main.rs` ([automatic](#automatic-project-setup) setup using cargo-casper), an empty `contract/main.rs` file ([manual](#manual-project-setup) project setup), or a rust "hello world" program defined in your `contract/main.rs` ([semi-automatic](#semi-automatic-project-setup) setup using "vanilla cargo"). In the following, you will write a new contract step-by-step. Therefore it is recommended to clear the content of contract/main.rs (if any).
```

> At this point you either have the default example contract defined in `contract/main.rs` ([automatic](#automatic-project-setup) setup using cargo-casper), an empty `contract/main.rs` file ([manual](#manual-project-setup) project setup), or a rust "hello world" program defined in your `contract/main.rs` ([semi-automatic](#semi-automatic-project-setup) setup using "vanilla cargo"). In the following, you will write a new contract step-by-step. Therefore it is recommended to clear the content of contract/main.rs (if any).

Reviewer: The correct paths are:

```suggestion
At this point you either have the default example contract defined in `contract/src/main.rs` ([automatic](#automatic-project-setup) setup using cargo-casper), an empty `contract/src/main.rs` file ([manual](#manual-project-setup) project setup), or a rust "hello world" program defined in your `contract/src/main.rs` ([semi-automatic](#semi-automatic-project-setup) setup using "vanilla cargo"). In the following, you will write a new contract step-by-step. Therefore it is recommended to clear the content of contract/main.rs (if any).
```

        EntryPointType::Contract,

>     ));

Reviewer: I think we should add the '}' at the end as well. This way user can really just copy-paste the snippet

```suggestion
    ));
}
```

        Vec::new(),
        CLType::I32,
        EntryPointAccess::Public,
        EntryPointType::Contract,
    ));

    counter_entry_points.add_entry_point(EntryPoint::new(
        ENTRY_POINT_COUNTER_INC,
        Vec::new(),
        CLType::Unit,
        EntryPointAccess::Public,
        EntryPointType::Contract,
    ));

    // Create a new contract package that can be upgraded
    let (stored_contract_hash, contract_version) = storage::new_contract(
        counter_entry_points,
        Some(counter_named_keys),
        Some(CONTRACT_PACKAGE_NAME.to_string()),
        Some(CONTRACT_ACCESS_UREF.to_string()),
    );

    /* To create a locked contract instead, use new_locked_contract and throw away the contract version returned
    let (stored_contract_hash, _) =
        storage::new_locked_contract(counter_entry_points, Some(counter_named_keys), None, None); */

    // Store the contract version in the context's named keys
    let version_uref = storage::new_uref(contract_version);
    runtime::put_key(CONTRACT_VERSION_KEY, version_uref.into());

    // Create a named key for the contract hash
    runtime::put_key(CONTRACT_KEY, stored_contract_hash.into());

> }

Reviewer: The function is correct, but if you follow the tutorial step by step the order is slightly different. Below is the order according to the tutorial an including the comment regarding the locked contracts

```suggestion
pub extern "C" fn call() {
    // Initialize the count to 0 locally
    let count_start = storage::new_uref(0_i32);
    // Create the entry points for this contract
    let mut counter_entry_points = EntryPoints::new();

    counter_entry_points.add_entry_point(EntryPoint::new(
        ENTRY_POINT_COUNTER_GET,
        Vec::new(),
        CLType::I32,
        EntryPointAccess::Public,
        EntryPointType::Contract,
    ));

    counter_entry_points.add_entry_point(EntryPoint::new(
        ENTRY_POINT_COUNTER_INC,
        Vec::new(),
        CLType::Unit,
        EntryPointAccess::Public,
        EntryPointType::Contract,
    ));

    // In the named keys of the counter contract, add a key for the count.
    let mut counter_named_keys = NamedKeys::new();
    let key_name = String::from(COUNT_KEY);
    counter_named_keys.insert(key_name, count_start.into());

    // Create a new contract package that can be upgraded.
    let (stored_contract_hash, contract_version) = storage::new_contract(
        counter_entry_points,
        Some(counter_named_keys),
        Some(CONTRACT_PACKAGE_NAME.to_string()),
        Some(CONTRACT_ACCESS_UREF.to_string()),
    );

    /* To create a locked contract instead, use new_locked_contract and throw away the contract version returned
    let (stored_contract_hash, _) =
    storage::new_locked_contract(counter_entry_points, Some(counter_named_keys), None, None); */

    // Store the contract version in the context's named keys.
    let version_uref = storage::new_uref(contract_version);
    runtime::put_key(CONTRACT_VERSION_KEY, version_uref.into());

    // Create a named key for the contract hash.
    runtime::put_key(CONTRACT_KEY, stored_contract_hash.into());
}
```

## Compiling Contract Code {#compiling-contract-code}

To compile the smart contract, run the following commands in the `contract` folder in your project's directory where `Cargo.toml` file and `src` folder are hosted.

```bash
rustup target add wasm32-unknown-unknown
cargo build --release --target wasm32-unknown-unknown
```

> For the counter example, in your project's directory where you have 'Makefile' run the following:

Reviewer: Changed quote to backtick for consistent formatting:

```suggestion
For the counter example, in your project's directory where you have `Makefile` run the following:
```

> To compile the smart contract, run the following command in the directory hosting the `Cargo.toml` file and `src` folder.

Reviewer: To be more precise:
```suggestion
To compile the smart contract, run the following commands in the `contract` folder in your project's directory where `Cargo.toml` file and `src` folder are hosted.
```

Reviewer: Because we also have `Cargo.toml` file and `src` folder in the 'tests' folder.

```bash
rustup target add wasm32-unknown-unknown
cargo build --release --target wasm32-unknown-unknown
```

> For the counter example, you may use the Makefile provided:

Reviewer: ```suggestion
For the counter example, in your project's directory where you have 'Makefile' run the following:
```
